### PR TITLE
HPCC-25680 Use cert-manager generated key for Roxie encryption in transit

### PR DESCRIPTION
**Pull request by afishbeck**
Signed-off-by: Anthony Fishbeck <anthony.fishbeck@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by afishbeck** [comment]
@richardkchapman here's a draft PR for getting the UDP key.

To install, go through the walkthrough in helm/examples/certmanager/README.md.

Still a work in progress, especially where the init of the udp key in jsecrets.cpp is concerned.

**Comment by afishbeck** [comment]
@richardkchapman converted from draft to PR.

File: system/jlib/jsecrets.cpp [hunk 1]
```diff
@@ -719,30 +722,39 @@ extern jlib_decl bool getSecretValue(StringBuffer & result, const char *category
...
     BIO_free(in);
 #endif
-    return ret;
+    udpKeyInitialized = true;
+}
+
+const MemoryAttr &getSecretUdpKey(bool required)
+{
+    if (!udpKeyInitialized)
+        throw makeStringException(-1, "UPD Key not initialized.");
```
Comment (richardkchapman) [inline system/jlib/jsecrets.cpp:754]: Is this supposed to say UDP ?

File: system/jlib/jsecrets.cpp [hunk 2]
```diff
@@ -719,30 +722,39 @@ extern jlib_decl bool getSecretValue(StringBuffer & result, const char *category
...
+    if (required && !udpKey.length())
+        throw makeStringException(-1, "UPD Key not found, cert-manager integration/configuration required.");
```
Comment (richardkchapman) [inline system/jlib/jsecrets.cpp:756]: And this?

**Review by richardkchapman — CHANGES_REQUESTED** [review]
A couple of typos.

What's the plan for bare metal? No encryption support?

File: system/jlib/jsecrets.cpp (hunk 1 above)
Comment (afishbeck) [reply system/jlib/jsecrets.cpp:754]: My fingers always seem to like this combination better.  Will change.

**Comment by afishbeck** [comment]
@richardkchapman updated to fix typo.

**Comment by afishbeck** [comment]
@richardkchapman @ghalliday 

This can be used in bare-metal if the roxie setting"encryptInTransit" is set to true, and the key file (PEM format) is somehow deployed to each roxie node in the following location: ""/opt/HPCCSystems/secrets/certificates/udp/tls.key".  I can/will improve that by at least looking up what the secrets root directory should be for the current system.

**Comment by afishbeck** [comment]
Updated to not only build upd key loading routine for containerized, now requires OPENSSL 1.1.0+.
Also build certificate directory instead of hardcoding, which could improve use of UPD key in some bare-metal scenarios.

**Review by richardkchapman — APPROVED** [review]

**Comment by richardkchapman** [comment]
Is there/should there be a documentation Jira ?

**Review by ghalliday — APPROVED** [review]
looks good to me.

**Pull request merged**